rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,109 @@
+# OPSGenie Integration RFD
+## Required approvers
+
+Engineering:
+Product:
+Security:
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with OpsGenie. This plugin will differ from existing plugins by being part of the Teleport binary directly.
+
+## Success criteria
+Users are able to configure teleport to automatically create alerts in OpsGenie from access requests.
+Alerts were chosen over incidents in OpsGenie for high priority alerts that indicate a service interruption.
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call 
+
+## Configuration UX
+
+The plugin will be configured using a toml file containing the required to interact with both Teleport access and the OpsGenie API.
+
+```
+[teleport]
+addr = "example.com:3025" # Teleport Auth Server GRPC API address
```

Comment:
If the plugin is running in the same process as the auth server, why do we need to give it an addr? Won't this just make it harder to configure correctly?

Reply:
Removed these.

Reply:
@zmb3 @EdwardDowling I think we want to maintain the ability to run the plugin separately, not just as part of the auth server. Similar to how our other access plugins function. For example `teleport plugin opsgenie start`, `teleport plugin pagerduty start`, etc.

Reply:
If we end up using the `teleport.yaml` format, then I guess we should support the current `teleport start --config teleport.yaml` flow.